### PR TITLE
missing C++98 functions

### PR DESCRIPTION
**Pull request by phlptp**
### Summary

If merged this pull request will add some missing C and C++98 API functions including set/get Info on inputs and isValid functions for endpoints, publications, inputs, and filters in the C API

Fixes Issues #1211 and #1213 

### Proposed changes

- add isValid to the C++98 and C API for inputs, publications, endpoints, and filters
- add get/setInfo to the C++98 input objects.
- add a few test of the added functions

### CODE REVIEW

**Comment by phlptp** [comment]
Those methods take an interface handle,  there are a number of methods in the C++ interface for Federate which take and interface_handle that don't exist in the C or C++98 interface since they are redundant.

File: src/helics/shared_api_library/MessageFederate.h [hunk 1]
```diff
@@ -95,6 +95,15 @@ HELICS_EXPORT helics_endpoint helicsFederateGetEndpoint(helics_federate fed, con
  */
 HELICS_EXPORT helics_endpoint helicsFederateGetEndpointByIndex(helics_federate fed, int index, helics_error* err);
 
+/**
+ * check if an endpoint is valid
+ *
+ * @param endpoint The endpoint object to check.
+ *
+ * @return helics_true if the Endpoint object represents a valid endpoint
```
Comment (nightlark) [inline src/helics/shared_api_library/MessageFederate.h:103]:
```suggestion
 * Check if an endpoint is valid.
 *
 * @param endpoint The endpoint object to check.
 *
 * @return helics_true if the Endpoint object represents a valid endpoint.
```

File: src/helics/shared_api_library/MessageFilters.h [hunk 2]
```diff
@@ -153,6 +153,15 @@ HELICS_EXPORT helics_filter helicsFederateGetFilter(helics_federate fed, const c
  */
 HELICS_EXPORT helics_filter helicsFederateGetFilterByIndex(helics_federate fed, int index, helics_error* err);
 
+/**
+ * check if a filter is valid
+ *
+ * @param filt The filter object to check.
+ *
+ * @return helics_true if the Filter object represents a valid filter
```
Comment (nightlark) [inline src/helics/shared_api_library/MessageFilters.h:161]:
```suggestion
 * Check if a filter is valid.
 *
 * @param filt The filter object to check.
 *
 * @return helics_true if the Filter object represents a valid filter.
```

File: src/helics/shared_api_library/ValueFederate.h [hunk 3]
```diff
@@ -296,6 +296,15 @@ HELICS_EXPORT void helicsFederatePublishJSON(helics_federate fed, const char* js
  * @{
  */
 
+ /**
+  * check if a publication is valid
+  *
+  * @param pub The publication to check.
+  *
+  * @return helics_true if the publication is a valid publication
```
Comment (nightlark) [inline src/helics/shared_api_library/ValueFederate.h:304]:
```suggestion
  * Check if a publication is valid.
  *
  * @param pub The publication to check.
  *
  * @return helics_true if the publication is a valid publication.
```

File: src/helics/shared_api_library/ValueFederate.h [hunk 4]
```diff
@@ -421,6 +430,15 @@ HELICS_EXPORT void helicsPublicationPublishNamedPoint(helics_publication pub, co
  */
 HELICS_EXPORT void helicsPublicationAddTarget(helics_publication pub, const char* target, helics_error* err);
 
+/**
+ * check if an input is valid
+ *
+ * @param ipt The input to check.
+ *
+ * @return helics_true if the Input object represents a valid input
```
Comment (nightlark) [inline src/helics/shared_api_library/ValueFederate.h:438]:
```suggestion
 * Check if an input is valid.
 *
 * @param ipt The input to check.
 *
 * @return helics_true if the Input object represents a valid input.
```

**Comment by phlptp** [comment]
@afisher1  if you come across anything else that you see as missing in the next couple days let us know.  I am going to merge this.

**Pull request merged**